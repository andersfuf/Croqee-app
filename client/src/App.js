import React, { Component } from 'react';
import './App.css';
import PrototypePage from './components/pages/prototype/PrototypePage';
import LoginPage from './components/pages/login/LoginPage';
import SignUpPage from './components/pages/signup/SignUpPage';
import LogoutFunction from './components/child/logout/LogoutFunction';
import { BrowserRouter as Router, Route, Switch, Link, Redirect, withRouter } from 'react-router-dom';
import Auth from './modules/Auth';
import NavBar from './components/child/navbar/NavBar';

const PrivateRoute = ({ component: Component, ...rest }) => (
	<Route
		{...rest}
		render={(props) =>
			Auth.isUserAuthenticated() ? (
				<Component {...props} {...rest} />
			) : (
				<React.Fragment>
					<Redirect
						to={{
							pathname: '/login',
							state: { from: props.location }
						}}
					/>
				</React.Fragment>
			)}
	/>
);

const LoggedOutRoute = ({ component: Component, ...rest }) => (
	<Route
		{...rest}
		render={(props) =>
			Auth.isUserAuthenticated() ? (
				<Redirect
					to={{
						pathname: '/',
						state: { from: props.location }
					}}
				/>
			) : (
				<Component {...props} {...rest} />
			)}
	/>
);
const GlobalRoute = ({ component: Component, ...rest }) => (
	<Route {...rest} render={(props) => <Component {...props} {...rest} />} />
);

class App extends Component {
	state = {};

	componentDidMount() {}

	render() {
		return (
			<div className="App">
				<Router>
					<NavBar />
					<div>
						<GlobalRoute exact path="/" component={PrototypePage} />
						<LoggedOutRoute path="/signup" component={SignUpPage} />
						<LoggedOutRoute path="/login" component={LoginPage} />
						<PrivateRoute path="/logout" component={LogoutFunction} />
					</div>
				</Router>
			</div>
		);
	}
}

export default App;
