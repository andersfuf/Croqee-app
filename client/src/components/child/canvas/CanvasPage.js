import React from 'react';
import axios from 'axios';
import { connect } from "react-redux";
import { setTimer, invokeScore, setImageProcessing } from '../../../js/actions';
import Loader from '../loader/Loader';

const styles = {
  canvas: {
    border: '1px solid #333',
    cursor: 'crosshair',
    // position: 'absolute',
    // left: '60px',
    // opacity: .8,
    // marginRight:'50px'
  },

  maindiv: {
    padding: '10px',
    // margin: 'auto 60px auto auto',
    width: '800px',
    // marginRight:'10px'

  },

  button: {
    border: '0px',
    // margin: '1px',
    height: '50px',
    minWidth: '75px',
  },

  colorSwatches: {
    red: { backgroundColor: 'red' },
    orange: { backgroundColor: 'orange' },
    yellow: { backgroundColor: 'yellow' },
    green: { backgroundColor: 'green' },
    blue: { backgroundColor: 'blue' },
    purple: { backgroundColor: 'purple' },
    black: { backgroundColor: 'black' },
  },
};

class CanvasPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    }
  }

  componentDidMount() {

    this.reset();
    this.sendDrawing();



  }

  draw(e) {
    //response to Draw button click
    this.setState({
      mode: 'draw',
    });
  }

  drawing(e) {
    //if the pen is down in the canvas, draw/erase

    if (this.state.pen === 'down') {
      this.ctx.beginPath();
      this.ctx.lineWidth = this.state.lineWidth;
      this.ctx.lineCap = 'round';

      if (this.state.mode === 'draw') {
        this.ctx.strokeStyle = this.state.penColor;
      }

      if (this.state.mode === 'erase') {
        this.ctx.strokeStyle = '#ffffff';
      }

      this.ctx.moveTo(this.state.penCoords[0], this.state.penCoords[1]); //move to old position
      this.ctx.lineTo(e.nativeEvent.offsetX, e.nativeEvent.offsetY); //draw to new position
      this.ctx.stroke();

      this.setState({
        //save new position
        penCoords: [e.nativeEvent.offsetX, e.nativeEvent.offsetY],
      });
    }
  }

  penDown(e) {
    //mouse is down on the canvas
    this.setState({
      pen: 'down',
      penCoords: [e.nativeEvent.offsetX, e.nativeEvent.offsetY],
    });
  }

  penUp() {
    //mouse is up on the canvas
    this.setState({
      pen: 'up',
    });
  }

  reset() {
    //clears it to all white, resets state to original
    this.setState({
      mode: 'draw',
      pen: 'up',
      lineWidth: 1.6,
      penColor: 'black',
    });
    if (this.refs.canvas) {
      this.ctx = this.refs.canvas.getContext('2d');
      this.ctx.fillStyle = 'white';
      this.ctx.fillRect(0, 0, 800, 600);
      this.ctx.lineWidth = 10;
    }
  }

  sendDrawing() {
    // var canvas = this.refs.canvas.getContext('2d');
    this.props.setTimer(true);
    setInterval(() => {
      var canvas = document.getElementById('canvas__drawing');
      console.log(canvas)

      if (canvas) {
        this.props.setImageProcessing(true);
        var dataURL = canvas.toDataURL().replace(/^data:image\/(png|jpg);base64,/, "");
        console.log(dataURL)
        this.props.setTimer(false);
        axios.post('/send_drawing', { dataURL: dataURL }).then(response => {
          // console.log(response);
          let { score } = response.data;
          score = score || 0;


          this.props.setTimer(true);
          this.props.setImageProcessing(false);
          this.props.invokeScore(score);
          this.reset();
        })
      }
    }, this.props.timer * 1000)
  }


  render() {
    let side = this.props.leftHand ? "canvas_left_hand" : ""
    return (
      /* We should separate this to another component (Canvas) for modularity reasons. But as we are using but we can't use the'ref' attribute
             in the functional components. We have to figure a way out later
            */
      <>

        <span id="userScore" className={this.props.scoreClass}>Score: {this.props.currentScore && this.props.currentScore}</span>
        <div className={"canvas " + side} style={styles.maindiv}>
          {this.props.imageProcessing && <Loader />}
          <span id="drawhere" />

          <canvas
            id="canvas__drawing"
            className="canvas__canvas "
            ref="canvas"
            width="800px"
            height="600px"
            style={styles.canvas}
            onMouseMove={e => this.drawing(e)}
            onMouseDown={e => this.penDown(e)}
            onMouseUp={e => this.penUp(e)}
          />
        </div>
      </>
    );
  }
}


const mapStateToProps = state => {
  const { timer, showTimer, scoreClass, currentScore, imageProcessing, leftHand } = state;
  return { timer, showTimer, scoreClass, currentScore, imageProcessing, leftHand };

};
const mapDispatchToProps = dispatch => {
  return {
    setTimer: (payload) => dispatch(setTimer(payload)),
    invokeScore: (payload) => dispatch(invokeScore(payload)),
    setImageProcessing: (payload) => dispatch(setImageProcessing(payload))

  };
}
export default connect(mapStateToProps, mapDispatchToProps)(CanvasPage);
