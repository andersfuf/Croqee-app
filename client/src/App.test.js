import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import CanvasPage from './components/canvas/CanvasPage';
import { configure ,shallow, mount, render  } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

describe('App', () => {
  it('renders correctly with canvas', () => {
    const div = document.createElement('div');
    const wrapper = render(<App />, div);
    expect(wrapper.find('.canvas').length).toBe(1);
  });
});

test('CanvasPage matches snapshot', () => {
  const div = document.createElement('div');
  const wrapper = render(<CanvasPage />, div);
  expect(wrapper).toMatchSnapshot();
});
