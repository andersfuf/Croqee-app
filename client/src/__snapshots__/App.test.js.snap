// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasPage matches snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "canvas",
      "style": "padding:10px;margin:auto;width:800px",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "canvas__title",
        },
        "children": Array [
          Object {
            "data": "Draw something",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h4",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "canvas__canvas",
            "height": "600px",
            "style": "border:1px solid #333;margin:20px 0px;cursor:crosshair",
            "width": "800px",
          },
          "children": Array [],
          "name": "canvas",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "height": undefined,
            "style": undefined,
            "width": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "height": undefined,
            "style": undefined,
            "width": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "canvas__canvas",
          "height": "600px",
          "style": "border:1px solid #333;margin:20px 0px;cursor:crosshair",
          "width": "800px",
        },
        "children": Array [],
        "name": "canvas",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "canvas__title",
          },
          "children": Array [
            Object {
              "data": "Draw something",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h4",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "height": undefined,
          "style": undefined,
          "width": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "height": undefined,
          "style": undefined,
          "width": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
